jdaviz/configs/imviz/helper.py

```diff
@@ -241,14 +241,36 @@ def load_data(self, data, data_label=None, do_link=True, show_in_viewer=True, **
                                                     "be removed in a future release.  Use with "
                                                     "viz.batch_load() instead."))
 
-    def link_data(self, **kwargs):
+    def link_data(self, link_type='pixels', wcs_fallback_scheme=None, wcs_use_affine=True):
         """(Re)link loaded data in Imviz with the desired link type.
         All existing links will be replaced.
 
-        See :func:`~jdaviz.configs.imviz.helper.link_image_data`
-        for available keyword options and more details.
+        Parameters
+        ----------
+        link_type : {'pixels', 'wcs'}
+            Choose to link by pixels or WCS.
+
+        wcs_fallback_scheme : {None, 'pixels'}
+            If WCS linking failed, choose to fall back to linking by pixels or not at all.
+            This is only used when ``link_type='wcs'``.
+            Choosing `None` may result in some Imviz functionality not working properly.
+
+        wcs_use_affine : bool
+            Use an affine transform to represent the offset between images if possible
+            (requires that the approximation is accurate to within 1 pixel with the
+            full WCS transformations). If approximation fails, it will automatically
+            fall back to full WCS transformation. This is only used when ``link_type='wcs'``.
+            Affine approximation is much more performant at the cost of accuracy.
+
         """
-        link_image_data(self.app, **kwargs)
+        from jdaviz.configs.imviz.plugins.orientation.orientation import link_type_msg_to_trait
+        plg = self.plugins["Orientation"]
+        plg._obj.linking_in_progress = True
+        plg.link_type = link_type_msg_to_trait[link_type]
+        plg._obj.wcs_use_fallback = wcs_fallback_scheme == 'pixels'
+        plg.wcs_use_affine = wcs_use_affine
+        plg._obj.linking_in_progress = False
+        plg._obj._update_link()
```

Comment:
I haven't been able to track down exactly what's wrong, but what you wrote can be simplified a bit, since changing `orientation.link_type` triggers all of the other necessary calls:

```suggestion
        from jdaviz.configs.imviz.plugins.orientation.orientation import link_type_msg_to_trait
        plg = self.plugins["Orientation"]
        plg._obj.wcs_use_fallback = wcs_fallback_scheme == 'pixels'
        plg.wcs_use_affine = wcs_use_affine
        plg.link_type = link_type_msg_to_trait[link_type]
```

Reply:
Are you sure the other traitlet assignments won't trigger the call, causing race conditions? I see an `observe` on all three of them.

Reply:
I accepted your suggestion and tests still passed, so probably ok. Thanks!